### PR TITLE
support vertex_credentials filepath

### PR DESCRIPTION
**Pull request by hawktang**
## Title

Implement user authentication feature

fix issue to read vertex_credentials

## Relevant issues

fix issue to read vertex_credentials

## Type

🐛 Bug Fix

## Changes

json_obj = jsons.load(vertex_credentials)

to

json_obj = json.load(open(vertex_credentials))

reference
https://google-auth.readthedocs.io/en/stable/reference/google.oauth2.service_account.html

```python
service_account_info = json.load(open('service_account.json'))
credentials = service_account.Credentials.from_service_account_info(
    service_account_info)
```

## [REQUIRED] Testing - Attach a screenshot of any new tests passing local

```python
from litellm import completion
import os

# os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "sia-data-team-vertexai-gcp-key_new.json"

model = "claude-3-opus@20240229"

vertex_ai_project = "..." # can also set this as os.environ["VERTEXAI_PROJECT"]
vertex_ai_location = "us-east5" # can also set this as os.environ["VERTEXAI_LOCATION"]

response = completion(
    model="vertex_ai/" + model,
    messages=[{"role": "user", "content": "hi"}],
    temperature=0.7,
    vertex_ai_project=vertex_ai_project,
    vertex_ai_location=vertex_ai_location,
    vertex_ai_credentials =  "./vertexai-gcp-key_new.json"
)
print("\nModel Response", response)
```
VertexAIError: Expecting value: line 1 column 1 (char 0)

image

image

change code, rerun script

### CODE REVIEW

File: litellm/llms/vertex_ai_anthropic.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def completion(
             if vertex_credentials is not None and isinstance(vertex_credentials, str):
                 import google.oauth2.service_account
 
-                json_obj = json.loads(vertex_credentials)
+                json_obj = json.load(open(vertex_credentials))
```
Comment (ghost) [inline litellm/llms/vertex_ai_anthropic.py]: hey @hawktang this assumes the vertex_credentials is a filepath 

That's not the intended purpose of this code block. It's a json str of the vertex_credentials 

I believe we can support both flows - can you change this to check if it's a filepath or json string and proceed accordingly

**Review by ghost — CHANGES_REQUESTED** [review]

File: litellm/llms/vertex_ai_anthropic.py (hunk 1 above)
Comment (ghost) [reply litellm/llms/vertex_ai_anthropic.py]: example for intended flow - 

https://github.com/BerriAI/litellm/blob/056913fd7049923a106130b02d7c29e7f312beec/litellm/tests/test_amazing_vertex_completion.py#L130

Comment (ghost) [reply litellm/llms/vertex_ai_anthropic.py]: Looking further at the example - do you even need to set this? 

If you have `GOOGLE_APPLICATION_CREDENTIALS` in your env, the vertex ai sdk should automatically be handling this?

**Comment by hawktang** [comment]
Hi @krrishdholakia, I submitted a new commit to fit support both flows, which can handle filepath or json string and proceed accordingly

```python
try:
    json_obj = json.loads(vertex_credentials)
except json.JSONDecodeError:
    json_obj = json.load(open(vertex_credentials))
```

When vertex_credentials is a JSON string
image

When vertex_credentials is a path
image

**Review by hawktang — COMMENTED** [review]
I have add the adaptor as requested

**Comment by ghost** [comment]
hey @hawktang can you fix the linting error - LGTM to merge after that!

**Comment by hawktang** [comment]
Hi @krrishdholakia,

Thank you so much, however it is not a linting error.

It is before the flake8 for lint, error is for poetry lock file regeneration. 

image

**Pull request merged**